pipework

```diff
@@ -205,43 +205,43 @@ else
 fi
 
 # Check if an incompatible VLAN device already exists
-[ $IFTYPE = phys ] && [ "$VLAN" ] && [ -d /sys/class/net/$IFNAME.VLAN ] && {
-  [ -z "$(ip -d link show $IFNAME.$VLAN | grep "vlan.*id $VLAN")" ] && {
+[ "$IFTYPE" = phys ] && [ "$VLAN" ] && [ -d "/sys/class/net/$IFNAME.VLAN" ] && {
+  ip -d link show "$IFNAME.$VLAN" | grep -q "vlan.*id $VLAN" || {
     echo "$IFNAME.VLAN already exists but is not a VLAN device for tag $VLAN"
     exit 1
   }
 }
 
 [ ! -d /var/run/netns ] && mkdir -p /var/run/netns
-[ -f /var/run/netns/$NSPID ] && rm -f /var/run/netns/$NSPID
-ln -s /proc/$NSPID/ns/net /var/run/netns/$NSPID
+rm -f "/var/run/netns/$NSPID"
```

Comment:
`rm -f` doesn't give a failure exit status or an error message when its argument isn't found, so it's not necessary to check.